test/Microsoft.ML.OnnxTransformTest/DnnImageFeaturizerTest.cs

```diff
@@ -0,0 +1,140 @@
+            var pipe = new DnnImageFeaturizerEstimator(Env, DnnImageFeaturizerEstimator.DnnImageModel.ResNet18, "data_0", "output_1");
+
+            var invalidDataWrongNames = ComponentCreation.CreateDataView(Env, xyData);
+            var invalidDataWrongTypes = ComponentCreation.CreateDataView(Env, stringData);
+            var invalidDataWrongVectorSize = ComponentCreation.CreateDataView(Env, sizeData);
+            TestEstimatorCore(pipe, dataView, invalidInput: invalidDataWrongNames);
+            TestEstimatorCore(pipe, dataView, invalidInput: invalidDataWrongTypes);
+            pipe.GetOutputSchema(SchemaShape.Create(invalidDataWrongVectorSize.Schema));
+            try
+            {
+                pipe.Fit(invalidDataWrongVectorSize);
+                Assert.False(true);
+            }
+            catch (ArgumentOutOfRangeException) { }
+            catch (InvalidOperationException) { }
+        }
+
+        [ConditionalFact(typeof(Environment), nameof(Environment.Is64BitProcess))] // x86 fails with "An attempt was made to load a program with an incorrect format."
+        public void OnnxStatic()
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                return;
+
+            using (var env = new ConsoleEnvironment(null, false, 0, 1, null, null))
+            {
+                var imageHeight = 224;
+                var imageWidth = 224;
+                var dataFile = GetDataPath("images/images.tsv");
+                var imageFolder = Path.GetDirectoryName(dataFile);
+
+                var data = TextLoader.CreateReader(env, ctx => (
+                    imagePath: ctx.LoadText(0),
+                    name: ctx.LoadText(1)))
+                    .Read(dataFile);
+
+                // Note that CamelCase column names are there to match the TF graph node names.
+                var pipe = data.MakeNewEstimator()
+                    .Append(row => (
+                        row.name,
+                        data_0: row.imagePath.LoadAsImage(imageFolder).Resize(imageHeight, imageWidth).ExtractPixels(interleaveArgb: true)))
+                    .Append(row => (row.name, output_1: row.data_0.DnnImageFeaturizer(DnnImageFeaturizerEstimator.DnnImageModel.ResNet18)));
+
+                TestEstimatorCore(pipe.AsDynamic, data.AsDynamic);
+
+                var result = pipe.Fit(data).Transform(data).AsDynamic;
+                result.Schema.TryGetColumnIndex("output_1", out int output);
+                using (var cursor = result.GetRowCursor(col => col == output))
+                {
+                    var buffer = default(VBuffer<float>);
+                    var getter = cursor.GetGetter<VBuffer<float>>(output);
+                    var numRows = 0;
+                    while (cursor.MoveNext())
+                    {
+                        getter(ref buffer);
+                        Assert.Equal(512, buffer.Length);
```

Comment:
Need more strict asserts like checking the prediction values.

Reply:
See the new OldSavingAndLoading test which contains some exact output value checks.

---
In reply to: [231254842](https://github.com/dotnet/machinelearning/pull/1447#discussion_r231254842) [](ancestors = 231254842)